common/iterator/sorted/IntersectForwardableIterator.java

```diff
@@ -159,10 +159,15 @@ public T next() {
 
     @Override
     public void forward(T target) {
+        if (state == State.COMPLETED) return;
         iterators.forEach(iterator -> iterator.forward(target));
```

Comment:
I think this line needs to be under a check as well. If A is in `[(1,2), >(2,3), …]`, and B is in `[>(1,3), …]`, we should still be able to forward to `(1,3)`. Instead, this line will attempt to advance A from `(2,3)` to `(1,3)`, triggering an exception.

Reply:
i see what you mean - if we're partway through an intersection containing many values (eg a large of tuples all starting with `(1, X))`, but some of the iterators then ready to start producing `(2, X)`. Forwarding them to `(1,X)` is a valid no-op, but will cause an exception in the iterators already placed at `(2,X)`!